### DISCUSSION TITLE
Join two shortcode files in a single variable

### DISCUSSION THREAD

**Discussion by rica-carv** [e107inc/e107#5257] [General]
How can i do it?

Imagine i need the code inside two diferent shortcode files, to parse to a single template, how can i join the two shortcode files?

**Comment by Vodhin** [comment]
The standard shortcodes are in a class, So I think you just need to include the other file and extend the class? not sure... I would probably just put the needed code in a function outside of a class, then just call that function in the shortcode file.

**Reply by rica-carv** [reply]
Some 5 years i go i tryed that, with earlyer v2.x, and was unable to make it work. Maybe i go for a try again now. PHP is diferent, so i might make it work this time... because i need it to make it work...

**Reply by Jimmi08** [reply]
@rica-carv it will not work for core shortcodes. Actually, you can write shortcodes 3-4 different ways (without shortcode file too, see sitelinks class, news etc). 
The main issue was complicated class naming, but it should be now solved and easier...

**Reply by rica-carv** [reply]
@Jimmi08 My issue is that i'm splitting code, shortcodes and templates on my plugin, but i have shortcodes that are used along the whole plugin pages, but others not, so since i've embarqued on this quest, right now i have no turning back and got to realize a way to do this shortcodes merging....

**Reply by Jimmi08** [reply]
@rica-carv use e_shortcode.php for global shortcodes.

**Reply by rica-carv** [reply]
> @rica-carv use e_shortcode.php for global shortcodes.

But that way it will be global for the entire site, right? I just want them "global" in my plugin, i don't need them available in the whole site...

**Reply by rica-carv** [reply]
Just managed to do it....
Include and expands is working now...

**Reply by Vodhin** [reply]
Great! Help others out by posting an example of what worked for you: you might just be left in someone's will later on because you were so nice 😄
